src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,111 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int DEFAULT_AGE = 27;
+    private static final int OTHER_AGE = 22;
+    private static final int WRONG_AGE1 = 0;
+    private static final int WRONG_AGE2 = -10;
+    private static final String DEFAULT_LOGIN = "login";
+    private static final String DEFAULT_PASSWORD = "password";
+    private static final String OTHER_PASSWORD = "another_Password";
+    private static final String WRONG_PASSWORD = "1234";
+    private static RegistrationServiceImpl regService;
+    private User defaultUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        regService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        defaultUser = new User();
+        defaultUser.setLogin(DEFAULT_LOGIN);
+        defaultUser.setPassword(DEFAULT_PASSWORD);
+        defaultUser.setAge(DEFAULT_AGE);
+    }
+
+    @Test
+    void register_DefaultUser_OK() {
+        User expected = defaultUser;
+        User actual = regService.register(defaultUser);
+        assertEquals(expected.getLogin(), actual.getLogin());
+        assertEquals(expected.getPassword(), actual.getPassword());
+        assertEquals(expected.getAge(), actual.getAge());
+    }
+
+    @Test
+    void register_nullUser_notOK() {
+        assertThrows(RuntimeException.class, () -> regService.register(null));
+    }
+
+    @Test
+    void register_idNotNull_notOK() {
+        User userIdNotNull = defaultUser;
+        userIdNotNull.setId(1L);
+        assertThrows(RuntimeException.class, () -> regService.register(userIdNotNull));
+    }
+
+    @Test
+    void register_NullLogin_notOK() {
```

Comment:
```suggestion
    void register_nullLogin_notOk() {
```
And everywhere below